File: apps/web/lib/schemas/lexicon.ts
import { z } from "zod";

export const LexiconEntrySchema = z.object({
  id: z.string(),
  surface: z.string(),
  language: z.string(),
  phonemes: z.string(),
  notes: z.string(),
  created_at: z.string(),
  updated_at: z.string(),
});
export type LexiconEntry = z.infer<typeof LexiconEntrySchema>;

export const LexiconListResponseSchema = z.object({
  entries: z.array(LexiconEntrySchema),
});
export type LexiconListResponse = z.infer<typeof LexiconListResponseSchema>;

export const LexiconCreateRequestSchema = z.object({
  surface: z.string().min(1),
  language: z.string().min(1),
  phonemes: z.string().min(1),
  notes: z.string().default(""),
});
export type LexiconCreateRequest = z.infer<typeof LexiconCreateRequestSchema>;

export const LexiconUpdateRequestSchema = z.object({
  phonemes: z.string().optional(),
  notes: z.string().optional(),
});
export type LexiconUpdateRequest = z.infer<typeof LexiconUpdateRequestSchema>;

File: apps/api/app/schemas/lexicon.py
from __future__ import annotations

from pydantic import BaseModel, Field


class LexiconEntrySchema(BaseModel):
    id: str
    surface: str
    language: str
    phonemes: str
    notes: str
    created_at: str
    updated_at: str


class LexiconListResponse(BaseModel):
    entries: list[LexiconEntrySchema]


class LexiconCreateRequest(BaseModel):
    surface: str = Field(min_length=1)
    language: str = Field(min_length=1)
    phonemes: str = Field(min_length=1)
    notes: str = ""


class LexiconUpdateRequest(BaseModel):
    phonemes: str | None = None
    notes: str | None = None
